test/Window/Context.test.cpp

```diff
@@ -1,8 +1,59 @@
 #include <SFML/Window/Context.hpp>
+#include <SFML/Window/ContextSettings.hpp>
 
+#include <doctest/doctest.h>
+
+#include <WindowUtil.hpp>
+#include <string>
 #include <type_traits>
 
 static_assert(!std::is_copy_constructible_v<sf::Context>);
 static_assert(!std::is_copy_assignable_v<sf::Context>);
 static_assert(!std::is_nothrow_move_constructible_v<sf::Context>);
 static_assert(!std::is_nothrow_move_assignable_v<sf::Context>);
+
+TEST_CASE("[Window] sf::Context" * doctest::skip(skipDisplayTests))
+{
+    SUBCASE("Construction")
+    {
+        const sf::Context context;
+
+        CHECK(context.getSettings().majorVersion > 0);
+    }
+
+    SUBCASE("Version String")
+    {
+        sf::Context context;
+
+        CHECK(context.setActive(true));
+
+#if defined(SFML_SYSTEM_WINDOWS)
+        using glGetStringFuncType = const char*(__stdcall*)(unsigned int);
+#else
+        using glGetStringFuncType = const char* (*)(unsigned int);
+#endif
```

Comment:
Seeing platform-related preprocessor conditionals in the tests makes me wonder if we're straying from the purpose of testing SFML's cross-platform API. If we have to do platform-specific things in the tests, that implies that users may have to do this too. Do we really need tests that are mostly just checking OpenGL implementation details? Do the existence of these tests hint at the need for an addition API for validation OpenGL details?

Reply:
A normal user would realistically never have to care about these specific details. The normal assumption is that they distribute their programs to people to run on hardware accelerated systems. In our case because we care about running in virtual CI environments as well, this is the simplest way for us to verify the OpenGL implementation is what we expect. The strings are dumped to the console before the tests are run.